conda_smithy/lint_recipe.py

```diff
@@ -1,6 +1,11 @@
 # -*- coding: utf-8 -*-
 
 from collections.abc import Sequence, Mapping
+from typing import List
+
+from pydantic import BaseModel
```

Comment:
@isuruf Let's continue the [discussion about adding pydantic](https://github.com/conda-forge/conda-smithy/pull/1900#discussion_r1578052953) here.

Reply:
By the way: `pydantic` is already listed as a project dependency in `environment.yml`.

Reply:
For clarity `pydantic` is a development-only dependency. So that only affects development, not installation. We generate a JSON schema for runtime use. For context, it was added in PR: https://github.com/conda-forge/conda-smithy/pull/1756

Is it possible to use `jsonschema` for this case too?

Reply:
I understand that pydantic is not currently a part of https://github.com/conda-forge/conda-smithy-feedstock/blob/main/recipe/meta.yaml dependencies.

However, I want to repeat my argument:

> I would like and strongly propose to keep pydantic here. Pydantic provides an elegant way to find additional fields > that are not part of the schema (see my implementation) - and also, I think the pydantic model should be used in > more parts of the smithy code anyway since it provides a type-safe way to access fields.


I am not aware of a good way doing this with `jsonschema`.

Reply:
Why is adding `pydantic` as a new dependency so bad?

Reply:
> Why is adding pydantic as a new dependency so bad?

Please see my comment on the other thread.

> I am not aware of a good way of doing this with jsonschema.

This is easy. Load the json schema and see if `"additionalProperties": false` is there for the attribute.

Reply:
> Please see my comment on the other thread.

@isuruf I just scrolled through every comment you left at my conda-smithy PRs but could not find an answer to my question. You said:

> This adds a new dependency pydantic on users. Previously it was only a dev dependency. Is it possible to remove this?

and 

> I strongly disagree. This PR use pydantic to check for which fields to hint about. If we can't do that using the json schema we are essentially making a distinction between pydantic model and json schema.

I do not see how this explains why we want to avoid `pydantic` as a dependency. To your latter comment, I already gave a response which I repeated above and elaborated why I see `pydantic` as the path forward for conda-smithy.

---

> This is easy. Load the json schema and see if "additionalProperties": false is there for the attribute.

This requires us to have a separate JSON schema that actually has `additionalProperties` set to false for the fields we want to warn about since we have agreed to allow additional properties in the main schema. Is this what you propose?

Reply:
@isuruf I would really really like to solve this debate as I want this PR to get merged.